src/open_inwoner/accounts/views/contacts.py

```diff
@@ -76,17 +77,23 @@ def get_form_kwargs(self):
 
     def form_valid(self, form):
         cleaned_data = form.cleaned_data
-        email = cleaned_data["email"]
+
         user = self.request.user
-        contact_user = User.objects.filter(email__iexact=email)
+        first_name = cleaned_data["first_name"]
+        last_name = cleaned_data["last_name"]
+        email = cleaned_data["email"]
+
+        contact_users = User.objects.filter(
```

Comment:
I would recommend to first do this based on only email, and only if there are duplicates to use first+last name to differentiate between them.

Reason being that otherwise I have to enter my contact's name exactly as they entered theirs.

Reply:
Discussed, this is done as described in forms.py. However we don't want to duplicate this code.

Recommendation is to keep the existing_user from the find_user() method and if not None then set this to the cleaned_data, this so the User can be reused from within form_valid method in the view.